### PR TITLE
Add Create and Delete HanaInstance commands to CLI

### PR DESCRIPTION
**Pull request by lagalbra**
- Take auto-generated files from azure-sdk-for-python
- Update release version
- Write connectors to call PutHanaInstance/DeleteHanaInstance from the CLI commands
    - Add necessary CLI parameters for Create to work

### CODE REVIEW

**Comment by lagalbra** [comment]
Testing performed as advised [here](https://microsoft.sharepoint.com/teams/AzureDedicatedEngineeringTeam/_layouts/OneNote.aspx?id=%2Fteams%2FAzureDedicatedEngineeringTeam%2FSiteAssets%2FAzure%20Dedicated%20Engineering%20Team%20Notebook&wd=target%28SAP%20HANA.one%7CFB7D0654-1276-104F-AF53-BA55FC4DA39E%2FAzure%20CLI%7C7303BF06-991D-4331-9BF8-542F65E5D86E%2F%29)

Logs showing attempts to create a new HanaInstance or delete an existing HanaInstance are rejected as not supported: https://jarvis-west.dc.ad.msft.net/19E1592D (I purposely did not test in a white-listed subscription where I would actually delete/create something that would mess up our runner). Deleting a non-existent resource returned a 204 from ARM and never queried our RP.

**Comment by lagalbra** [comment]
I will not merge this PR until https://github.com/Azure/azure-hanaonazure-cli-extension/pull/22 is merged; it fixes a breaking change, and sets a lower version of the extension than this PR does.

**Review by pabowers — APPROVED** [review]
LGTM

**Comment by lagalbra** [comment]
\<rebased against master\>

**Review by Caiyeon — APPROVED** [review]

**Review by AprilZhaoWY — APPROVED** [review]
LGTM

File: azext_hanaonazure/modules_sdk/models/hana_instance_py3.py [hunk 1]
```diff
@@ -78,10 +86,12 @@ class HanaInstance(Resource):
         'power_state': {'key': 'properties.powerState', 'type': 'str'},
         'proximity_placement_group': {'key': 'properties.proximityPlacementGroup', 'type': 'str'},
         'hw_revision': {'key': 'properties.hwRevision', 'type': 'str'},
+        'partner_node_id': {'key': 'properties.partnerNodeId', 'type': 'str'},
+        'provisioning_state': {'key': 'properties.provisioningState', 'type': 'str'},
     }
 
-    def __init__(self, *, hardware_profile=None, storage_profile=None, os_profile=None, network_profile=None, **kwargs) -> None:
-        super(HanaInstance, self).__init__(**kwargs)
+    def __init__(self, *, location: str=None, hardware_profile=None, storage_profile=None, os_profile=None, network_profile=None, partner_node_id: str=None, **kwargs) -> None:
```
Comment (persiaAziz-zz) [inline azext_hanaonazure/modules_sdk/models/hana_instance_py3.py:93]: I see all of these parameters have default value of None. So all of these are optional?

File: azext_hanaonazure/modules_sdk/operations/hana_instances_operations.py [hunk 2]
```diff
@@ -241,6 +241,193 @@ def get(
...
+        # Construct and send request
+        request = self._client.put(url, query_parameters, header_parameters, body_content)
+        response = self._client.send(request, stream=False, **operation_config)
+
+        if response.status_code not in [200, 201]:
+            raise models.ErrorResponseException(self._deserialize, response)
+
+        deserialized = None
+
+        if response.status_code == 200:
```
Comment (persiaAziz-zz) [inline azext_hanaonazure/modules_sdk/operations/hana_instances_operations.py:283]: Nit: these two conditions can be combined

File: azext_hanaonazure/modules_sdk/operations/hana_instances_operations.py [hunk 3]
```diff
@@ -241,6 +241,193 @@ def get(
...
+            if raw:
+                client_raw_response = ClientRawResponse(deserialized, response)
+                return client_raw_response
+
+            return deserialized
+
+        lro_delay = operation_config.get(
+            'long_running_operation_timeout',
+            self.config.long_running_operation_timeout)
+        if polling is True: polling_method = ARMPolling(lro_delay, **operation_config)
```
Comment (persiaAziz-zz) [inline azext_hanaonazure/modules_sdk/operations/hana_instances_operations.py:344]: formatting for conditional statements look unconventional. What tool are you using for formatting

File: azext_hanaonazure/custom.py [hunk 4]
```diff
@@ -6,6 +6,28 @@
 # pylint: disable=no-self-use,too-many-lines
 
 
+def create_hanainstance(client, location, resource_group_name, instance_name, ssh_public_key, os_computer_name, ip_address):
+    try:
```
Comment (persiaAziz-zz) [inline azext_hanaonazure/custom.py:10]: Excuse my limited knowledge on this extension codebase. Are we trying to import from python3 campatible version first and if we fail then we fall back to 2.7?

File: azext_hanaonazure/modules_sdk/models/hana_instance_py3.py (hunk 1 above)
Comment (lagalbra) [reply azext_hanaonazure/modules_sdk/models/hana_instance_py3.py:93]: Yes; it's up to us as an RP to reject things missing required data. From the CLI, we can specify required parameters and then fill out these structures like `hardware_profile`.

Note: everything under azext_hanaonazure/modules_sdk/* is auto-generated, not written by me

File: azext_hanaonazure/modules_sdk/operations/hana_instances_operations.py (hunk 2 above)
Comment (lagalbra) [reply azext_hanaonazure/modules_sdk/operations/hana_instances_operations.py:283]: Everything under azext_hanaonazure/modules_sdk/* is auto-generated, not written by me

File: azext_hanaonazure/modules_sdk/operations/hana_instances_operations.py (hunk 3 above)
Comment (lagalbra) [reply azext_hanaonazure/modules_sdk/operations/hana_instances_operations.py:344]: I couldn't tell you what formatting tool is used; this comes from https://github.com/Azure/azure-sdk-for-python/tree/master/sdk/hanaonazure/azure-mgmt-hanaonazure/azure/mgmt/hanaonazure

Everything under azext_hanaonazure/modules_sdk/* is auto-generated, not written by me

File: azext_hanaonazure/custom.py (hunk 4 above)
Comment (lagalbra) [reply azext_hanaonazure/custom.py:10]: Correct

**Comment by lagalbra** [comment]
@dwj300 Please review; waiting since you expressed a desire to review

**Comment by lagalbra** [comment]
I've successfully tested `create` and `delete` on these commits:

```
az hanainstance create --debug -n Test02 --ip-address 10.19.17.52 -l japaneast --os-computer-name Test02 --partner-node-id "/subscriptions/1ba60399-6ce1-4704-8155-5921d35e914e/resourceGroups/TYO-T520/providers/Microsoft.HanaOnAzure/hanaInstances/Test01" -g TYO-T520 --ssh-public-key "<a-key>"; echo $?
{
  "hanaInstanceId": "68e4e5df-52bd-4c4d-892f-af64f363cee2",
  "hardwareProfile": {
    "hanaInstanceSize": "S192",
    "hardwareType": "Cisco_UCS"
  },
  "hwRevision": "Rev 3",
  "id": "/subscriptions/1ba60399-6ce1-4704-8155-5921d35e914e/resourceGroups/TYO-T520/providers/Microsoft.HanaOnAzure/hanaInstances/Test02",
  "location": "japaneast",
  "name": "Test02",
  "networkProfile": {
    "circuitId": "NA",
    "networkInterfaces": [
      {
        "ipAddress": "10.19.17.52"
      }
    ]
  },
  "osProfile": {
    "computerName": "Test02",
    "osType": null,
    "sshPublicKey": "<a-key>",
    "version": null
  },
  "partnerNodeId": "/subscriptions/1ba60399-6ce1-4704-8155-5921d35e914e/resourceGroups/TYO-T520/providers/Microsoft.HanaOnAzure/hanaInstances/Test01",
  "powerState": "unknown",
  "provisioningState": "Succeeded",
  "proximityPlacementGroup": null,
  "resourceGroup": "TYO-T520",
  "storageProfile": {
    "nfsIpAddress": "10.180.211.51",
    "osDisks": null
  },
  "tags": {},
  "type": "Microsoft.HanaOnAzure/hanaInstances"
}



az hanainstance delete --debug -n Test02 -g TYO-T520
```

**Comment by lagalbra** [comment]
\<rebased against master>

**Comment by dwj300** [comment]
Lgtm

**Review by dwj300 — APPROVED** [review]

**Pull request merged**